cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -367,17 +370,25 @@ public Map<ApplicationId, ApplicationDetail> getAppDetails(Collection<Applicatio
     Set<? extends EntityId> visibleIds = accessEnforcer.isVisible(new HashSet<>(appIds),
                                                                   authenticationContext.getPrincipal());
     Set<ApplicationId> filterIds = appIds.stream().filter(visibleIds::contains).collect(Collectors.toSet());
-    Map<ApplicationId, ApplicationSpecification> appSpecs = store.getApplications(filterIds);
+    Map<ApplicationId, ApplicationMeta> appMetas = store.getApplications(filterIds);
```

Comment:
This part is also in https://github.com/cdapio/cdap/pull/14926. Will rebase once the mentioned fix PR gets merged